new-docs/search/main.go

```diff
@@ -0,0 +1,215 @@
+		logger.Error("There was an error walking the public directory", "error", err)
+		return nil, err
+	}
+	return fileList, nil
+}
+
+func loadConfiguration() bool {
+	if algoliaApplicationID == nil || *algoliaApplicationID == "" {
+		logger.Error("Missing Algolia Application ID", "algoliaApplicationID", algoliaApplicationID)
+		flag.Usage()
+		return false
+	}
+	if algoliaAdminAPIKey == nil || *algoliaAdminAPIKey == "" {
+		logger.Error("Missing Algolia Admin API Key", "algoliaAdminAPIKey", algoliaAdminAPIKey)
+		flag.Usage()
+		return false
+	}
+	if algoliaIndex == nil || *algoliaIndex == "" {
+		logger.Error("Missing Algolia Index", "algoliaIndex", algoliaIndex)
+		flag.Usage()
+		return false
+	}
+	if *verbose {
+		logger.SetLevel(hclog.Debug)
+		logger.Info("Setting logging level to DEBUG")
+	}
+	return true
+}
+
+func processFiles(args []string) (urls []Record, err error) {
+	uniqueRecords := make(map[string]*Record)
+
+	for _, item := range args {
+		contents, err := ioutil.ReadFile(item)
+
+		if err != nil {
+			logger.Error("Error loading file", "error", err)
+			return nil, err
+		}
+		logger.Info("Processing JSON file", "name", item, "length", len(contents))
+
+		doc, err := simplejson.NewJson(contents)
+		if err != nil {
+			logger.Error("Error processing JSON file", "error", err)
+			return nil, err
+		}
+
+		for _, val := range doc.MustArray() {
+			vals := val.(map[string]interface{})
+
+			permalink := vals["permalink"].(string)
+
+			if uniqueRecords[permalink] == nil {
+				record := &Record{
+					Description:  vals["description"].(string),
+					Kind:         vals["kind"].(string),
+					Language:     vals["lang"].(string),
+					Permalink:    permalink,
+					RecordWeight: ProcessRecordWeight(vals),
+					Section:      vals["section"].(string),
+					Summary:      vals["summary"].(string),
+					Title:        vals["title"].(string),
+				}
+				if record.RecordWeight > 0 {
+					uniqueRecords[permalink] = record
+				} else {
+					logger.Debug("Skipping", "record", record)
+				}
+			}
+
+		}
+	}
+	for _, record := range uniqueRecords {
+		urls = append(urls, *record)
+	}
+	return urls, nil
+}
+
+func ProcessRecordWeight(vals map[string]interface{}) int64 {
+	// we want to weight changelogs, so we will set their "kind"
+	permalink := vals["permalink"].(string)
+	if matchChangelog.MatchString(permalink) {
+		logger.Debug("Found a changelog")
+		// these will be included but at a low weight
+		return 1
+	}
+
+	kind := vals["kind"].(string)
+	section := vals["section"].(string)
+
+	logger.Debug("ProcessKind", "kind", kind, "section", section, "permalink", permalink)
+
+	switch kind {
+	case "home":
+		// this is the main home page and we will include it
+		return 10 // this means it isn't weighted - default is 10
+	case "taxonomy":
+		// this is a taxonomy page, like https://docs.osohq.com/node/tags.html, so we ignore it
+		return 0
+	case "page":
+		return 10
+	case "section":
+		switch section {
+		case "getting-started":
```

Comment:
Curious why we you have this case for weighing getting started sections to 0

Reply:
https://docs.osohq.com/rust/getting-started.html for example just holds links and a small amount of content for other pages.